beacon-chain/core/epoch/precompute/processing.go

```diff
@@ -0,0 +1,194 @@
+package precompute
+
+import (
+	"bytes"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+)
+
+// New gets called at the beginning of process epoch cycle to return
+// pre computed instances of validators attesting records and total
+// balances attested in an epoch.
+func New(state *pb.BeaconState) ([]*Validator, *Balance) {
```

Comment:
Same here, might want a span because it could be epxnesive

Reply:
Done